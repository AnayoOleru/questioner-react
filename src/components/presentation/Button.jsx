import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Button = ({
  type,
  children,
  role,
  onClick,
  className,
  disabled
}) => (
  <Fragment>
    <button
      type={type}
      role={role}
      onClick={onClick}
      className={className}
      disabled={disabled}
    >
      {children}
    </button>
  </Fragment>
);

export default Button;
