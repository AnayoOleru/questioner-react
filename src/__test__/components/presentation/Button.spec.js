import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/presentation/Button';

describe('Button Component', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      type: 'button',
      role: 'button',
      onClick: jest.fn(),
      disabled: false
    };
    wrapper = shallow(<Button {...props} />);
  });
  test('Button component should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
