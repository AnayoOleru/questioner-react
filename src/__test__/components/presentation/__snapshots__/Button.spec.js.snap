// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button component should render 1`] = `ShallowWrapper {}`;
